code/game/objects/items/devices/t_scanner.dm

```diff
@@ -9,27 +9,41 @@
 	item_state = "electronic"
 	matter = list(DEFAULT_WALL_MATERIAL = 150)
 	origin_tech = list(TECH_MAGNET = 1, TECH_ENGINEERING = 1)
+	action_button_name = "Toggle T-Ray scanner"
 
-	var/scan_range = 1
+	var/scan_range = 3
 
 	var/on = 0
 	var/list/active_scanned = list() //assoc list of objects being scanned, mapped to their overlay
 	var/client/user_client //since making sure overlays are properly added and removed is pretty important, so we track the current user explicitly
-	var/flicker = 0
 
 	var/global/list/overlay_cache = list() //cache recent overlays
 
+/obj/item/device/t_scanner/Destroy()
+	. = ..()
+	if(on)
+		set_active(FALSE)
+
 /obj/item/device/t_scanner/update_icon()
 	icon_state = "t-ray[on]"
 
+/obj/item/device/t_scanner/emp_act()
+	audible_message(src, "<span class = 'notice'> \The [src] buzzes oddly.</span>")
+	set_active(FALSE)
+
 /obj/item/device/t_scanner/attack_self(mob/user)
 	set_active(!on)
+	user.update_action_buttons()
+
+/obj/item/device/t_scanner/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	var/obj/structure/disposalpipe/D = target
+	if(D && istype(D))
+		to_chat(user, "<span class='info'>Pipe segment integrity: [(D.health / 10) * 100]%</span>")
```

Comment:
```suggestion
		to_chat(user, span("info", "Pipe segment integrity: [(D.health / 10) * 100]%"))
```

Reply:
in future change it to span proc

small change, but goes a long way in code cleanliness